hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/StreamSync.java

```diff
@@ -333,56 +334,56 @@ public StreamSync(HoodieStreamer.Config cfg, SparkSession sparkSession,
-      try {
-        HoodieTableMetaClient metaClient = getMetaClient();
-        switch (metaClient.getTableType()) {
-          case COPY_ON_WRITE:
-          case MERGE_ON_READ:
-            // we can use getCommitsTimeline for both COW and MOR here, because for COW there is no deltacommit
-            this.commitsTimelineOpt = Option.of(metaClient.getActiveTimeline().getCommitsTimeline().filterCompletedInstants());
-            this.allCommitsTimelineOpt = Option.of(metaClient.getActiveTimeline().getAllCommitsTimeline());
-            break;
-          default:
-            throw new HoodieException("Unsupported table type :" + metaClient.getTableType());
-        }
-      } catch (HoodieIOException e) {
-        LOG.warn("Full exception msg " + e.getMessage());
-        if (e.getMessage().contains("Could not load Hoodie properties") && e.getMessage().contains(HoodieTableConfig.HOODIE_PROPERTIES_FILE)) {
-          String basePathWithForwardSlash = cfg.targetBasePath.endsWith("/") ? cfg.targetBasePath :
-              String.format("%s/", cfg.targetBasePath);
-          String pathToHoodieProps = String.format("%s%s/%s", basePathWithForwardSlash,
-              HoodieTableMetaClient.METAFOLDER_NAME, HoodieTableConfig.HOODIE_PROPERTIES_FILE);
-          String pathToHoodiePropsBackup = String.format("%s%s/%s", basePathWithForwardSlash,
-              HoodieTableMetaClient.METAFOLDER_NAME,
-              HoodieTableConfig.HOODIE_PROPERTIES_FILE_BACKUP);
-          boolean hoodiePropertiesExists =
-              storage.exists(new StoragePath(basePathWithForwardSlash))
-                  && storage.exists(new StoragePath(pathToHoodieProps))
-                  && storage.exists(new StoragePath(pathToHoodiePropsBackup));
-          if (!hoodiePropertiesExists) {
-            LOG.warn("Base path exists, but table is not fully initialized. Re-initializing again");
-            initializeEmptyTable();
-            // reload the timeline from metaClient and validate that its empty table. If there are any instants found, then we should fail the pipeline, bcoz hoodie.properties got deleted by mistake.
-            HoodieTableMetaClient metaClientToValidate = getMetaClient();
-            if (metaClientToValidate.reloadActiveTimeline().countInstants() > 0) {
-              // Deleting the recreated hoodie.properties and throwing exception.
-              storage.deleteDirectory(new StoragePath(String.format("%s%s/%s", basePathWithForwardSlash,
-                  HoodieTableMetaClient.METAFOLDER_NAME,
-                  HoodieTableConfig.HOODIE_PROPERTIES_FILE)));
-              throw new HoodieIOException(
-                  "hoodie.properties is missing. Likely due to some external entity. Please populate the hoodie.properties and restart the pipeline. ",
-                  e.getIOException());
-            }
+  public HoodieTableMetaClient refreshTimeline() throws IOException {
+    HoodieTableMetaClient metaClient = null;
```

Comment:
Can we return Option instead of potential null ?

Reply:
Fixed, thanks for the review!

Reply:
Usually, we just initialize the metaClient with `Option.empty`.